### PR TITLE
Wait for GitHub to finish the "mergeability computation" before merging

### LINKED ISSUE #85: Trying to merge too fast?
Cleaned up logs from https://github.com/JuliaRegistries/General/runs/281762322#step:5:0:
```
[ Info: Now examining pull request 4890
[ Info: Pull request: 4890. Type: NewVersion. Decision: merge now.
[ Info: Attempting to squash-merge pull request #4890
┌ Debug: sha = 891fd7de6173ddbba186570f63d67d08f4da4f8d
└ @ RegistryCI.AutoMerge ~/.julia/packages/RegistryCI/x6ZH8/src/AutoMerge/github.jl:160

[ Info: Now examining pull request 4888
[ Info: Pull request: 4888. Type: NewVersion. Decision: merge now.
[ Info: Attempting to squash-merge pull request #4888
┌ Debug: sha = 9dedc9b57e660f82f2c5f3f7477be3712b36c14f
└ @ RegistryCI.AutoMerge ~/.julia/packages/RegistryCI/x6ZH8/src/AutoMerge/github.jl:160
Error found in GitHub reponse:
   Status Code: 405
   Message: Base branch was modified. Review and try the merge again.
   Docs URL: https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-button
   Errors: 
Stacktrace:
 [1] error(::String, ::String, ::String, ::String, ::String) at ./error.jl:42
 [2] handle_response_error(::HTTP.Messages.Response) at /home/runner/.julia/packages/GitHub/oQdYa/src/utils/requests.jl:159

[...]

[ Info: Now examining pull request 4814
[ Info: Pull request: 4814. Type: NewVersion. Decision: merge now.
[ Info: Attempting to squash-merge pull request #4814
┌ Debug: sha = 9e6239ac91bca4e0d165b7405d90f5460e6c3236
└ @ RegistryCI.AutoMerge ~/.julia/packages/RegistryCI/x6ZH8/src/AutoMerge/github.jl:160
```
So it looks like the second merge try was too fast after the first one, but then after examining some more PRs (`[...]`) it worked. It is a bit weird that it works on Travis, but maybe everything is a bit slower there and we were lucky?

### PR DESCRIPTION
**Pull request by DilumAluthge**
Fixes #85

### CODE REVIEW

File: src/AutoMerge/AutoMerge.jl [hunk 1]
```diff
@@ -23,6 +23,7 @@ include("cron.jl")
 include("github.jl")
 include("guidelines.jl")
 include("jll.jl")
+include("mergeable.jl")
```
Comment (fredrikekre) [inline src/AutoMerge/AutoMerge.jl]: put it in `github.jl` instead?

Comment (DilumAluthge) [reply src/AutoMerge/AutoMerge.jl]: Done.

File: src/AutoMerge/github.jl [hunk 2]
```diff
@@ -176,6 +178,21 @@ function merge!(registry_repo::GitHub.Repo,
     return nothing
 end
 
+function wait_pr_compute_mergeability(repo::GitHub.Repo,
+                                      pr::GitHub.PullRequest;
+                                      auth::GitHub.Authorization)
+    sleep(5)
+    max_tries = 10
+    num_tries = 0
+    pr = GitHub.pull_request(repo, pr.number; auth = auth)
```
Comment (fredrikekre) [inline src/AutoMerge/github.jl]: This call might not be needed if the PR is already mergeable?

Comment (DilumAluthge) [reply src/AutoMerge/github.jl]: So, here's the problem.

Suppose we have two PR's, PR1 and PR2.

We have a vector containing both of those PR's, and we iterate through the vector.

First we process PR1, and decide to merge it, so we merge it.

Merging PR1 might change whether or not PR2 is mergeable. But when we merge PR1, it doesn't update our local copy of the PR2 object. So we have to re-fetch PR2 from the GitHub API to see if PR2 is still mergeable after PR1 is merged.

Comment (fredrikekre) [reply src/AutoMerge/github.jl]: Yea, just realized.

**Pull request merged**